### PR TITLE
3.0 release

### PR DESCRIPTION
**Pull request by transistive**
Merges the PRS of @stefanak-michal, fully commits to version 8.0, improves testing and provides full support for Neo4j 5. Working draft.

### CODE REVIEW

File: src/Types/Node.php [hunk 1]
```diff
@@ -22,30 +23,22 @@
...
      * @param CypherList<string>  $labels
      * @param CypherMap<OGMTypes> $properties
      */
-    public function __construct(int $id, CypherList $labels, CypherMap $properties)
-    {
-        $this->id = $id;
-        $this->labels = $labels;
-        $this->properties = $properties;
-    }
+    public function __construct(
```
Comment (stefanak-michal) [inline src/Types/Node.php:37]: Do you know they added new field into structures Node, Relationship and Path? It's element_id. I see you don't have it there yet. https://www.neo4j.com/docs/bolt/current/bolt/structure-semantics/#structure-node

I don't know if it's intentional, I'll just point it out.

**Comment by stefanak-michal** [comment]
You are not updating dependency to bolt library? https://github.com/neo4j-php/neo4j-php-client/pull/165/files#diff-d2ab9925cad7eac58e0ff4cc0d251a937ecf49e4b6bf57f8b95aab76648a9d34R35

That can be also answer on why you didn't add element_id I've mentioned here https://github.com/neo4j-php/neo4j-php-client/pull/165#pullrequestreview-1267772313

**Comment by stefanak-michal** [comment]
> Merges the PRS of @stefanak-michal, fully commits to version 8.0, improves testing and provides full support for Neo4j 5. Working draft.

If you won't update bolt library dependency you will not have full support for ^5. Because they fixed some issue with DateTime and DateTimeZoneId structure and added elementId.

**Comment by transistive** [comment]
> You are not updating dependency to bolt library?
Of course I am going to update it to the newest version of bolt! I am in the process of modernizing everything at the moment :)

Thank for all the extra info

**Comment by stefanak-michal** [comment]
But it is getting more strange. Because you removed catch MessageException and IgnoreException in BoltConnection class methods, which was introduced in bolt library v5.0.0.

https://github.com/neo4j-php/neo4j-php-client/pull/165/files#diff-610165784cb2bec45d288f332ef70a80f3f7eb5c3c9e52743e0b13b010362f73L285-L295

File: src/Types/Node.php (hunk 1 above)
Comment (transistive) [reply src/Types/Node.php:37]: Good thinking! This is a very early draft but I'll make sure it is there

**Comment by stefanak-michal** [comment]
> > You are not updating dependency to bolt library?
> 
> Of course I am going to update it to the newest version of bolt! I am in the process of modernizing everything at the moment :)
> 
> Thank for all the extra info

okay .. i was getting worried 😄  otherwise good job 👍

**Pull request merged**